src/services/pcn-packetcapture/src/Packetcapture.cpp

```diff
@@ -0,0 +1,363 @@
+      t_in.set(0x0, ON_T);
+      t_out.set(0x0, OFF_T);
+      CapStatus = 0;
+    break;
+    case PacketcaptureCaptureEnum::EGRESS:
+      t_in.set(0x0, OFF_T);
+      t_out.set(0x0, ON_T);
+      CapStatus = 1;
+    break;
+    case PacketcaptureCaptureEnum::BIDIRECTIONAL:
+      t_in.set(0x0, ON_T);
+      t_out.set(0x0, ON_T);
+      CapStatus = 2;
+    break;
+    case PacketcaptureCaptureEnum::OFF:
+      t_in.set(0x0, OFF_T);
+      t_out.set(0x0, OFF_T);
+      CapStatus = 3;
+    break;
+  }
+}
+
+bool Packetcapture::getAnomimize() {
+  //TODO
+}
+
+void Packetcapture::setAnomimize(const bool &value) {
+  //TODO
+}
+
+std::string Packetcapture::getDump() {
+  std::stringstream dump;
+
+  if(network_mode_flag){
+    dump << "the service is running in network mode";
+  }else{
+      char *cwdr_ptr = get_current_dir_name();
+      std::string wdr(cwdr_ptr);
+      dump << "capture dump in " << wdr << "/capture_"+ dt +".pcap" << std::endl;
+  }
+  
+  return dump.str();
+}
+
+void Packetcapture::setDump(const std::string &value) {
+    throw std::runtime_error("Packetcapture::setDump: Method not implemented");
+}
+
+bool Packetcapture::getNetworkmode() {
+  return network_mode_flag;
+}
+
+void Packetcapture::setNetworkmode(const bool &value) {
+  network_mode_flag = value;
+}
+
+std::shared_ptr<Filters> Packetcapture::getFilters() {
+  return filters;
+}
+
+void Packetcapture::addFilters(const FiltersJsonObject &value) {
+  filters = std::shared_ptr<Filters>(new Filters(*this, value));
+}
+
+void Packetcapture::replaceFilters(const FiltersJsonObject &conf) {
+  PacketcaptureBase::replaceFilters(conf);
+}
+
+void Packetcapture::delFilters() {
+  throw std::runtime_error("Packetcapture::delFilters: method not implemented");
+}
+
+/* return the first packet in the queue if exist */
+std::shared_ptr<Packet> Packetcapture::getPacket() {
+  if(packets_captured.size() != 0){
+    auto p = packets_captured.front();
+    if(network_mode_flag){                            /* pop this element */
+      packets_captured.erase(packets_captured.begin());
+      if(filters->getSnaplen() < p->getPacketlen()){
+        p->packet.resize(filters->getSnaplen());
+      }
+    }
+    return p;
+  }
+  PacketJsonObject pj;
+  auto p = std::shared_ptr<Packet>(new Packet(*this, pj));
+  return p;
+}
+
+void Packetcapture::addPacket(const std::vector<uint8_t> &packet,
+    const packetHeaders &pkt_values) {
+    PacketJsonObject pj;
+    auto p = std::shared_ptr<Packet>(new Packet(*this, pj));
+    struct timeval tp;
+    gettimeofday(&tp, NULL);
+    p->setTimestampSeconds((uint32_t) tp.tv_sec);
+    p->setTimestampMicroseconds((uint32_t) tp.tv_usec);
+    p->setPacketlen((uint32_t) packet.size());
+    if(p->getPacketlen() > filters->getSnaplen()){
+      p->setCapturelen(filters->getSnaplen());
+    }else{
+      p->setCapturelen(p->getPacketlen());
+    }
+    p->setRawPacketData(packet);
+    packets_captured.push_back(p);
```

Comment:
just curious in networking mode if no client consumes the packet, how many packet will be captured in the list?

Reply:
currently there is no maximum size that limits the amount of stored packets.
This can be a problem for the memory used. 

We can try solutions like:

- Temporarily dump packets in the file system
- use structures such as ring buffers, admitting to having to eventually have a maximum number of packages in memory